webhooks-extension/src/components/WebhookDisplayTable/WebhookDisplayTable.js

```diff
@@ -134,48 +136,66 @@ export class WebhookDisplayTable extends Component {
+
+    Promise.all(
+      deletePromises.map(promise =>
+        promise.then(deletionSuccess).catch(deletionError)
+      ))
+      .then(resultsOfPromises => {
+        if (resultsOfPromises.indexOf(false) === -1 ) {
+          if (this.state.webhooks.length - rowsToUse.length === 0) {
+            this.props.setshowLastWebhookDeletedNotification(true);
+          } else {
+            this.setState({
+              showNotificationOnTable: true,
+              showDeleteDialog: false,
+              notificationStatus: "success",
+              notificationStatusMsgShort: "Webhook(s) deleted successfully.",
+              notificationMessage: ""
+            });
+          }
+        } else {
+          throw new Error();
+        }
+      })
+      .catch(() => {
+        let notificationStatusToSet = "";
+        let notificationMessageToSet = "";
+        let notificationFullMessage = "";
+
+        if (displayTimeoutWarningNotification) {
+          notificationStatusToSet = "warning";
+          notificationMessageToSet =
+            "Warning - timed out waiting to delete webhooks and potentially PipelineRuns: manually check the resources were deleted.";
+        } else {
+          notificationStatusToSet = "error";
+          notificationMessageToSet = "An error occurred deleting webhook(s).";
+          notificationFullMessage =
+            "Check the webhook(s) existed and both the dashboard and extension pods are healthy.";
+        }
         this.setState({
           showNotificationOnTable: true,
           showDeleteDialog: false,
-          notificationStatus: 'success',
-          notificationStatusMsgShort: 'Webhook(s) deleted successfully.',
-          notificationMessage: '',
+          notificationStatus: notificationStatusToSet,
+          notificationStatusMsgShort: notificationMessageToSet,
+          notificationMessage: notificationFullMessage
         });
-      }
-     }).catch( () => {
-      this.fetchWebhooksForTable();
-      let notificationStatusToSet = '';
-      let notificationMessageToSet = '';
-      let notificationFullMessage = '';
-
-      if (displayTimeoutWarningNotification) {
-        notificationStatusToSet = 'warning'
-        notificationMessageToSet = 'Warning - timed out waiting to delete webhooks and potentially PipelineRuns: manually check the resources were deleted.';
-      } else {
-        notificationStatusToSet = 'error'
-        notificationMessageToSet =  'An error occurred deleting webhook(s).';
-        notificationFullMessage = 'Check the webhook(s) existed and both the dashboard and extension pods are healthy.';
-      }
-      this.setState({
-        showNotificationOnTable: true,
-        showDeleteDialog: false,
-        notificationStatus: notificationStatusToSet,
-        notificationStatusMsgShort: notificationMessageToSet,
-        notificationMessage: notificationFullMessage,
+      })
+      .finally(() => {
+        this.state.batchActions.onCancel();
```

Comment:
wouldn't this still unselect any remaining rows in case of error?

Reply:
True, moving it inside `then()`.